projects/reliable-integration/0002-low-level-design.md

```diff
@@ -6,13 +6,49 @@ This low-level design intends to go deeper on the technical aspects of the proje
 
 ## Full-node execution
 
+Considering this full-node cycle:
+
 ![Full-Node-Cycle drawio](./0002-images/full_node_cycle.png)
 
-### TODO: This section will be updated in a separate PR
+At first, when the full node is initialized, it moves to the `LOAD` state, where data is loaded from the database, without consulting the network. Once all local data is read, the full-node continuously syncs with peers, changing to the `SYNC` state.
+
+In order to know which events will be generated, we will store the states (`LOAD` and `SYNC`) as metadata. So, when the full node starts, we will check the metadata. If it is empty or `LOAD`, we do not know where it stopped and will regenerate all events during load. If it is `SYNC`, we will not generate any events until `LOAD_FINISHED` is emitted.
+
+In the table below we describe all possible starting scenarios for the full node, and the respective desired outcome:
+
+| Starting scenario                 | Vertices in the database | Events in the database | State metadata | Desired outcome                                                                                                                           |
+|-----------------------------------|--------------------------|------------------------|----------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| `Fresh start`                     | None                     | None                   | Empty          | Generate all events, from `event_id = 0`. All events come during the `SYNC` phase.                                                        |
+| `Snapshot start`                  | Some                     | None                   | Empty          | Generate all events, from `event_id = 0`. Events come during both the `LOAD` and `SYNC` phases.                                           |
+| `Restart from aborted LOAD phase` | Some                     | Some                   | `LOAD`         | All events in the database are discarded. Generate all events, from `event_id = 0`. Events come during both the `LOAD` and `SYNC` phases. |
+| `Restart from aborted SYNC phase` | Some                     | Some                   | `SYNC`         | Generate events from `event_id = last_event_id + 1`. All events come during the `SYNC` phase.                                             |
+
+There's no case in which there are some events in the database, but no vertices.
+
+For the `Fresh start` scenario, if we consider there's a `LOAD` phase with 0 events, then the desired outcome becomes the same for the first three scenarios. This makes it possible to simplify the table into only two scenarios, only depending on `State metadata`, as described in the beginning of this section:
```

Comment:
Would we emit events for the genesis transactions (two transactions and one block)?

Reply:
I'm in favor of emitting events for genesis vertices on a fresh start instead of having 0 events.

Reply:
We would. Added in https://github.com/HathorNetwork/rfcs/pull/52/commits/3a6c667fdbf028613e51def7b455cadf80dec784

Reply:
Great! Thanks!